website/project/views/contributor.py

```diff
@@ -633,7 +633,6 @@ def claim_user_form(auth, **kwargs):
             # Redirect to CAS and authenticate the user with a verification key.
```

Comment:
There might be some length issues; see checklist below.

There are two basic rules for randomly generated tokens:

- All possibilities have to be equally likely, eg the (pseudo)RNG is good. This helper function uses `SystemRandom`, which is in line with what the [docs](https://docs.python.org/2/library/random.html#random.SystemRandom) recommend.

- The chances of guessing accepted credentials should be extremely small; this seems much shorter than the recommended limits accepted by security professionals elsewhere. As one metric for comparison, RFC6749 (for OAuth) [suggests](https://tools.ietf.org/html/rfc6749#section-10.10):
> The probability of an attacker guessing generated tokens (and other
   credentials not intended for handling by end-users) MUST be less than
   or equal to 2^(-128) and SHOULD be less than or equal to 2^(-160).

Some back of the envelope calculations: In this case, the character set of the function (letters + numbers) = 62 characters to choose from, so the number of possible combinations is something like `(62)**20 ~= (64)**20 = (2**6)**20 = (2**120)`- a 20-character string falls 256x short of the minimum randomness.

I'd recommend a longer string- 30 characters long at least, and often we use longer. For OAuth client secrets, for example, [we use](https://github.com/abought/osf.io/blob/6c26058b338712fb159b4993b85c74a9cb13aa20/website/oauth/models/__init__.py#L470-L470) a 40-character string, which is to say `~= (2**6)**40 = 2**240`. For access tokens, I think the string is even longer- 70 characters.